miner/worker.go

```diff
@@ -547,60 +547,75 @@ func (w *worker) mainLoop() {
 		}
 	}()
 
+	logTime := func(workType string, action func()) {
+		start := time.Now()
+		defer func() {
+			elapsed := time.Since(start)
+			log.Trace("Completed work", "type", workType, "elapsed", common.PrettyDuration(elapsed))
+		}()
+		action()
+	}
+
 	for {
 		select {
 		case req := <-w.newWorkCh:
-			w.commitWork(req.interrupt, req.timestamp)
+			logTime("newWork", func() {
+				w.commitWork(req.interrupt, req.timestamp)
+			})
 
 		case req := <-w.getWorkCh:
-			req.result <- w.generateWork(req.params)
+			logTime("getWork", func() {
+				req.result <- w.generateWork(req.params)
+			})
 
 		case ev := <-w.txsCh:
-			if w.chainConfig.Optimism != nil && !w.config.RollupComputePendingBlock {
-				continue // don't update the pending-block snapshot if we are not computing the pending block
-			}
-			// Apply transactions to the pending state if we're not sealing
-			//
-			// Note all transactions received may not be continuous with transactions
-			// already included in the current sealing block. These transactions will
-			// be automatically eliminated.
-			if !w.isRunning() && w.current != nil {
-				// If block is already full, abort
-				if gp := w.current.gasPool; gp != nil && gp.Gas() < params.TxGas {
-					continue
+			logTime("txsCh", func() {
+				if w.chainConfig.Optimism != nil && !w.config.RollupComputePendingBlock {
+					return // don't update the pending-block snapshot if we are not computing the pending block
```

Comment:
I applied your logging patch in https://github.com/ethereum-optimism/op-geth/pull/187 and then changed it so that there's less diff (no indentation).

This made me realize that this line got changed from a `continue` to a `return` statement. Was that intentional?